Classes/common/CDTReplicator/CDTReplicator.m

```diff
@@ -100,6 +100,39 @@ - (id)initWithTDReplicatorManager:(TDReplicatorManager *)replicatorManager
 
 - (void)dealloc { [[NSNotificationCenter defaultCenter] removeObserver:self]; }
 
+- (NSString*) cleanRemote
+{
+    NSURL *remote = self.tdReplicator.remote;
+    return [NSString stringWithFormat:@"%@://%@:****@%@%@", remote.scheme, remote.user,
+        remote.host, remote.path ];
+}
+
+- (NSString*) description
+{
+    NSString *replicationType;
+    NSString *source;
+    NSString *target;
+    
+    if ( self.tdReplicator.isPush ) {
+        replicationType = @"push" ;
+        source = self.tdReplicator.db.name;
+        target = [self cleanRemote];
+        
+    }
+    else {
+        replicationType = @"pull" ;
+        source = [self cleanRemote];
+        target = self.tdReplicator.db.name;
+    }
+    
+    NSString *fullinfo = [NSString stringWithFormat: @"CDTReplicator %@, source: %@, target: %@ "
+                          @"filter name: %@, filter parameters %@, unique replication session "
+                          @"ID: %@", replicationType, source, target, self.tdReplicator.filterName,
+                          self.tdReplicator.filterParameters, self.tdReplicator.sessionID];
+    
+    return fullinfo;
+}
+
```

Comment:
I think all of this should be moved into https://github.com/cloudant/CDTDatastore/pull/104